### ISSUE TITLE
suggestions about importing library

### ISSUE THREAD

**Issue by chitrakgupta** [datacarpentry/python-ecology-lesson#218]
Just a thought that when introducing the concept of importing a python library, it might be worthwhile to discuss the different ways of importing, i.e.

1. import library
2. import library as lb
3. from library import function

Especially the 3rd one is helpful when you only need one or two functions from a library, in which case you need not import the entire library.

**Comment by wrightaprilm** [comment]
All right. Are you looking to make a pull request with that content, or just to have me add it to the to-do list?

**Comment by chitrakgupta** [comment]
Actually whichever is easier. I could go ahead and make these changes.

**Comment by wrightaprilm** [comment]
Either way is fine with me. I think it's good experience to merge at least one commit with us and see the feedback process.

**Comment by alexmorley** [comment]
I'm not 100% sure this (i.e. `from x import y` syntax) should be introduced straight away. The only reason I say this is to explain the real difference between it and the first two is quite subtle and could be confusing for someone just starting out. 

So `from x import y` still imports the whole module but just introduces `y` into the global namespace - (which also introduces the chance of a collision. I feel like there are quite a few concepts (modules, variable binding & namespaces) to introduce at once. And there's nothing that you can't do without knowing this bit.

The use of having it in might be that then a learner will know what it means if they find it in some code they are reading, but then in my opinion it's easy enough to google/look up in a reference manual.

Just my thoughts, feel free to ignore :smile:

**Comment by wrightaprilm** [comment]
Right, I agree with that. There are lots of opportunities in the materials to introduce something like that, either within the lessons proper or the capstone, which is why I favor @chitrakgupta  thinking about it and then making the pull request. I think this is one where there's no one single right place for it, and I'm looking forward to the discussion.

**Comment by brownsarahm** [comment]
This lesson as is introduces the first two and uses `import pandas as pd` throughout, so it's a change to make throughout to not introduce that.  It can be used without explaining all of the details of namespaces and I think it's valuable to do so because it is standard practice.  No one is going to find code excerpts elsewhere using `pandas`, `numpy` or `matplotlib.pylab` spelled out fully anywhere, so even without the full CS explanation for why it works, I think the nickname version is the best one to use.

**Comment by alexmorley** [comment]
@brownsarahm sorry if I didn't explain very well but I also meant to say I think it best to keep it as it is at the moment. Was just suggesting not to add the `from x import y` syntax (at least at the start) which AFAICT is not used in the lessons at present.

**Comment by tdpearson** [comment]
There is a similar discussion that is captured in the Software Carpentry's Instructor notes for the [Programming with Python](http://swcarpentry.github.io/python-novice-inflammation/guide/) lesson. They have valid arguments both for and against including the import x as y.

**Comment by chitrakgupta** [comment]
Thanks @tdpearson for referring me to the post. It was really interesting. I guess it is a pretty divided community on this issue. Since I don't have any instructor experience yet I won't be able to comment. I am presuming such details would depend on the how things go at a particular workshop? i.e. since python usually doesn't come up until the afternoon session, maybe take a decision based on how the morning session went?

**Comment by wrightaprilm** [comment]
I think that's a fine approach. I often adjust content over lunch or while the other instructor is talking based on how the audience is responding to material. I like going in with import pandas as pd right away, since when participants google for help, this is the syntax they'll see most often. I think the last session is fairly rarely taught, since DC (unlike SWC) has required curricula in all four time slots during a workshop. So I would guess that the people running that are people who wanted more hands-on and more depth after a workshop. That might be a place to put it. Adding a note in the instructor guide is another approach, and puts the material somewhere it can be used at the discretion of the instructor.
